src/scipp/plotting/plot3d.py

```diff
@@ -35,9 +36,16 @@ def builder(*,
                 show_outline=True,
                 xlabel=None,
                 ylabel=None,
-                zlabel=None):
+                zlabel=None,
+                camera=None):
         array = next(iter(scipp_obj_dict.values()))
         pos = array.meta[positions] if isinstance(positions, str) else positions
+        if camera is not None:
+            for k, v in camera.items():
+                if v.unit != pos.unit:
```

Comment:
I would say that conversions between compatible units should be allowed. For example, you could supply a position in `mm` if the positions are defined in `m`.
We could use `to_unit()` to do the conversion and this would raise the error if the units are incompatible?